plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,457 @@
+	return sampleConfig
+}
+
+func (n *NFSClient) Description() string {
+	return "Read per-mount NFS metrics from /proc/self/mountstats"
+}
+
+func convertToInt64(line []string) []int64 {
+	var nline []int64
+	for _, l := range line[1:] {
+		val, err := strconv.ParseInt(l, 10, 64)
+		if err != nil {
+			if numError, ok := err.(*strconv.NumError); ok {
+				if numError.Err == strconv.ErrRange {
+					log.Fatalf("ErrRange: %v [%v] -> [%v]\n", line, l, val)
+				}
+			}
+		}
+		nline = append(nline, val)
+	}
+	return nline
+}
+
+func in(list []string, val string) bool {
+	for _, v := range list {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+func (n *NFSClient) parseStat(mountpoint string, export string, version string, line []string, fullstat bool, nfs3Ops map[string]bool, nfs4Ops map[string]bool, acc telegraf.Accumulator) error {
+
+	tags := map[string]string{"mountpoint": mountpoint, "serverexport": export}
+	nline := convertToInt64(line)
+
+	if len(nline) == 0 {
+		n.Log.Errorf("Parsing Stat line didn't return enough fields(%d): %s\n", len(nline), line)
+		return nil
+	}
+
+	first := strings.Replace(line[0], ":", "", 1)
+
+	var eventsFields = []string{
+		"inoderevalidates",
+		"dentryrevalidates",
+		"datainvalidates",
+		"attrinvalidates",
+		"vfsopen",
+		"vfslookup",
+		"vfsaccess",
+		"vfsupdatepage",
+		"vfsreadpage",
+		"vfsreadpages",
+		"vfswritepage",
+		"vfswritepages",
+		"vfsgetdents",
+		"vfssetattr",
+		"vfsflush",
+		"vfsfsync",
+		"vfslock",
+		"vfsrelease",
+		"congestionwait",
+		"setattrtrunc",
+		"extendwrite",
+		"sillyrenames",
+		"shortreads",
+		"shortwrites",
+		"delay",
+		"pnfsreads",
+		"pnfswrites",
+	}
+
+	var bytesFields = []string{
+		"normalreadbytes",
+		"normalwritebytes",
+		"directreadbytes",
+		"directwritebytes",
+		"serverreadbytes",
+		"serverwritebytes",
+		"readpages",
+		"writepages",
+	}
+
+	var xprtudpFields = []string{
+		"bind_count",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var xprttcpFields = []string{
+		"bind_count",
+		"connect_count",
+		"connect_time",
+		"idle_time",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var nfsopFields = []string{
+		"ops",
+		"trans",
+		"timeouts",
+		"bytes_sent",
+		"bytes_recv",
+		"queue_time",
+		"response_time",
+		"total_time",
+		"errors",
+	}
+
+	var fields = make(map[string]interface{})
+
+	if first == "READ" {
+		fields["read_ops"] = nline[0]
+		fields["read_retrans"] = nline[1] - nline[0]
+		fields["read_bytes"] = nline[3] + nline[4]
+		fields["read_rtt"] = nline[6]
+		fields["read_exe"] = nline[7]
+		acc.AddFields("nfsstat_read", fields, tags)
+	} else if first == "WRITE" {
+		fields["write_ops"] = nline[0]
+		fields["write_retrans"] = nline[1] - nline[0]
+		fields["write_bytes"] = nline[3] + nline[4]
+		fields["write_rtt"] = nline[6]
+		fields["write_exe"] = nline[7]
+		acc.AddFields("nfsstat_write", fields, tags)
+	}
```

Comment:
You now write two measurements here, so please remove the `read_` and `write_` prefix in the field-names. Additionally you could write to only a `nfsstat` measurement and add an `operation` tag that is `read` or `write`...
```suggestion
	if first == "READ" || first == "WRITE" {
		fields["ops"] = nline[0]
		fields["retrans"] = nline[1] - nline[0]
		fields["bytes"] = nline[3] + nline[4]
		fields["rtt"] = nline[6]
		fields["exe"] = nline[7]
		tags["operation"] = strings.ToLower(first)
		acc.AddFields("nfsstat", fields, tags)
	}
```